contracts/modules/legacy/LegacyPooledStaking.sol

```diff
@@ -1207,213 +1208,196 @@ contract LegacyPooledStaking is IPooledStaking, MasterAwareV2 {
-
-    // Addresses marked for implicit migration can be migrated by anyone.
-    // Addresses who are not can only be migrated by calling this function themselves.
+    // Only selected stakers are automatically migrated
     require(
-      stakerAddress == ARMOR || // Armor
-      stakerAddress == HUGH || // Hugh
-      stakerAddress == NEXUS_FOUNDATION, // Foundation
+      stakerAddress == ARMOR_STAKER ||
+      stakerAddress == HUGH ||
+      stakerAddress == NM_FOUNDATION,
 
       "You are not authorized to migrate this staker"
     );
 
+    // Read and set deposit to zero to avoid re-entrancy
+    uint deposit = stakers[stakerAddress].deposit;
+    require(deposit > 0, "Address has no stake to migrate");
+    stakers[stakerAddress].deposit = 0;
 
     INXMToken nxm = token();
     uint nxmBalanceBefore = nxm.balanceOf(address(this));
 
-    // ratios have no decimal points. eg 5 is 5%
-    uint[TRANCHE_COUNT] memory stakerTrancheRatios;
-
-    (ProductInitializationParams[] memory params, uint deposit) = getStakerConfig(stakerAddress);
+    ProductInitializationParams[] memory productInitParams = getProductInitParams(
+      stakerAddress,
+      deposit
+    );
 
     if (stakerAddress == HUGH) {
-      stakerTrancheRatios = [uint256(0), 10, 0, 0, 0, 90, 0, 0];
-
       migrateToPool(
         StakingPoolMigrationData(
-         10, // initialPoolFee
-         20, // maxPoolFee
-         stakerTrancheRatios,
-         deposit,
-         HUGH,
-         false, // isPrivatePool
-         '', // ipfsDescriptionHash
-         HUGH // managerAddress
+          HUGH, // stakerAddress
+          HUGH, // managerAddress
+          '', // ipfsDescriptionHash --- TODO fill in
+          false, // isPrivatePool
+          10, // initialPoolFee
+          20, // maxPoolFee
+          deposit, // deposit
+          [uint256(0), 10, 0, 0, 0, 90, 0, 0] // stake on each tranche, as % out of the deposit
         ),
-        params
+        productInitParams
       );
-    } else if (stakerAddress == ARMOR) {
-      stakerTrancheRatios = [uint256(20), 25, 25, 15, 10, 0, 0, 0];
+    } else if (stakerAddress == ARMOR_STAKER) {
 
-      uint aaaLowRiskPoolDeposit = 75 * deposit / 100;
-      uint maxFee = 25;
-      uint initialFee = 15;
+      uint armorAAALowRiskPoolDeposit = 75 * deposit / 100;
       migrateToPool(
         StakingPoolMigrationData(
-          initialFee,
-          maxFee,
-          stakerTrancheRatios,
-          aaaLowRiskPoolDeposit,
-          ARMOR,
+          ARMOR_STAKER, // stakerAddress
+          ARMOR_MANAGER, // managerAddress
+          '', // ipfsDescriptionHash --- TODO fill in
           false, // isPrivatePool
-          '', // ipfsDescriptionHash
-          ARMOR_MANAGER // managerAddress
+          15, // initialPoolFee
+          25, // maxPoolFee
+          armorAAALowRiskPoolDeposit, // deposit
+          [uint256(20), 25, 25, 15, 10, 0, 0, 0] // stake on each tranche, as % out of the deposit
```

Comment:
this array is the same as the one below for Armor, i could argue for using the same constant as it already was

Reply:
I am trying to avoid black boxes and scrolling back and forth. it doesn't make a difference afterall